draft/0005-impersonation.md

```diff
@@ -0,0 +1,260 @@
+---
+Start date: 2024-05-30
+Author(s): wangsijie
+Revision: 1
+---
+
+# Impersonation
+
+## Overview
+
+User impersonation is a feature that allows an authorized user to assume the identity of another user within the system. This capability can be useful for various reasons such as troubleshooting, providing support, and performing administrative tasks. This document aims to explore the design and implementation of user impersonation in Logto.
+
+## Motivation
+
+Logto is an enterprise ready identity provider, and it is essential to provide user impersonation capabilities to support various use cases. For example, a support engineer may need to troubleshoot an issue on behalf of a user, or an administrator may need to verify the permissions of a user. User impersonation can help to streamline these processes and improve the overall user experience.
+
+## Token exchange
+
+The user impersonation feature will be implemented using the OAuth 2.0 [token exchange flow](https://www.rfc-editor.org/rfc/rfc8693.html). The token exchange flow allows an authorized user to exchange their access token for a new access token that represents the identity of another user. The new access token can then be used to make API requests on behalf of the impersonated user.
+
+### Request
```

Comment:
how about an actor token?  So the actor claim can present in the returned impersonate access_token

Reply:
Updated